relayer/submit_transaction.go

```diff
@@ -119,3 +137,68 @@ func checkForMinClaimBalance(ctx context.Context, ec extethclient.EthClient) (*b
 
 	return gasPrice, nil
 }
+
+// simulateExecute calls the forwarder's execute method (defined in Forwarder.sol)
+// with CallContract which executes the method call without mining it into the blockchain.
+// https://pkg.go.dev/github.com/ethereum/go-ethereum/ethclient#Client.CallContract
+func simulateExecute(
+	ctx context.Context,
+	ec extethclient.EthClient,
+	reqForwarderAddr *ethcommon.Address,
+	txOpts *bind.TransactOpts,
+	forwarderReq gsnforwarder.IForwarderForwardRequest,
+	domainSeparator [32]byte,
+	sig []byte,
+) ([]byte, error) {
+	ForwarderABI, err := abi.JSON(strings.NewReader(gsnforwarder.ForwarderMetaData.ABI))
```

Comment:
You can simplify that with `gsnforwarder.ForwarderMetaData.GetAbi()`? I would use a lower-case letter for a local variable, unless it confirms to some style guide that you think is useful and want to convince us to adopt.

Reply:
Nice, thank you